src/components/FeatureStyler/FeatureStyler.js

```diff
@@ -497,6 +614,21 @@ class FeatureStyler extends PureComponent {
       textRotation = featStyle.getText().getRotation();
     }
 
+    const oldStyles = FeatureStyler.getStyleAsArray(feature);
+
+    const newLineStartIcon =
+      lineStartIcon ||
+      FeatureStyler.findLineIcon(oldStyles, feature.getGeometry(), lineIcons);
```

Comment:
`findLineIcon` function updates the state of the component from the current style of the feature. This function doesn't care of the current state of the component. Imagine you click on another line the state of the component must be updated according to the new feature's style and ignoring what is the current state.

That's why you should remove the 'lineStartIcon || ' line.

Reply:
Done.